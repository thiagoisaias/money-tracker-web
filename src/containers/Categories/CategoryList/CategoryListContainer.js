import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import { fetchCategories } from "../../../store/actions/categories/categories";

import CategoryList from "../../../components/Categories/CategoryList/CategoryList";

export class CategoryListContainer extends Component {
  render() {
    const { categoryList, error, isLoading } = this.props;
    return (
      <CategoryList
        categoryList={categoryList}
        error={error}
        isLoading={isLoading}
      />
    );
  }

  componentDidMount() {
    const { onFetchCategoryList } = this.props;
    onFetchCategoryList();
  }
}

CategoryListContainer.propTypes = {
  categoryList: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      color: PropTypes.string.isRequired
    })
  ).isRequired,
  error: PropTypes.string,
  isLoading: PropTypes.bool.isRequired,
  onFetchCategoryList: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  return {
    categoryList: state.categories.categoryList,
    error: state.categories.error,
    isLoading: state.categories.isLoading
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onFetchCategoryList: () => {
      dispatch(fetchCategories());
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(
  CategoryListContainer
);
