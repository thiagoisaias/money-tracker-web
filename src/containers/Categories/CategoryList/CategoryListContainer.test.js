import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { CategoryListContainer } from "./CategoryListContainer";

describe("CategoryListContainer", () => {
  const mockProps = {
    categoryList: [
      { id: 1, name: "Home", color: "red" },
      { id: 2, name: "Food", color: "blue" }
    ],
    error: null,
    isLoading: false,
    onFetchCategoryList: jest.fn()
  };
  const component = shallow(<CategoryListContainer {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
