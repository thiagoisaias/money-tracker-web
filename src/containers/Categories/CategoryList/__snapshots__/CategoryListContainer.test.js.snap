// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryListContainer renders properly 1`] = `
<Component
  categoryList={
    Array [
      Object {
        "color": "red",
        "id": 1,
        "name": "Home",
      },
      Object {
        "color": "blue",
        "id": 2,
        "name": "Food",
      },
    ]
  }
  error={null}
  isLoading={false}
/>
`;
